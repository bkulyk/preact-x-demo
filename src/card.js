import React from 'react';
import PropTypes from 'prop-types';
import Typography from './typography';

const Card = ({ title, subtitle, content, media, height, width, children }) => (
  <div className="mdc-card" style={{ height, width }}>
    <div className="mdc-card__primary-action">
      {media}
    </div>
    <div className="demo-card__primary" style={{ padding: '0px 16px' }}>
      <Typography use="headline6" className="card__title" component="h2">{title}</Typography>
      <Typography use="subtitle2" className="card__subtitle" component="h3">{subtitle}</Typography>
    </div>
    <Typography className="demo-card__secondary" use="body2" style={{ padding: '0px 16px 8px 16px' }} component="div">
      {content}
    </Typography>
    {children}
  </div>
);

Card.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string,
  content: PropTypes.node,
  width: PropTypes.number,
  height: PropTypes.number,
  media: PropTypes.node,
  children: PropTypes.node,
};

Card.defaultProps = {
  title: '',
  subtitle: '',
  content: '',
  width: 350,
  height: 'auto',
  media: '',
  children: '',
};

export default Card;
