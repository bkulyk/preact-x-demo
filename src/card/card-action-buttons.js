import React from 'react';
import { node } from 'prop-types';

const CardActionButtons = ({ ...props }) => (
  <div {...props} className="mdc-card__action-buttons" />
);

CardActionButtons.propTypes = {
  children: node.isRequired,
};

export default CardActionButtons;
