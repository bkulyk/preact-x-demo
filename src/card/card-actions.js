import React from 'react';

const CardActions = ({ ...props }) => (
  <div {...props} className="mdc-card__actions" />
);

export default CardActions;
