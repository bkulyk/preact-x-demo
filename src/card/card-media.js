import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ title, url, ratio }) => (
  <div className="mdc-card__primary-action">
    <div className={`mdc-card__media mdc-card__media--${ratio}`} style={{ backgroundImage: `url(${url})` }}>
      <div className="mdc-card__media-content">{title}</div>
    </div>
  </div>
);

Card.propTypes = {
  title: PropTypes.string,
  url: PropTypes.string.isRequired,
  ratio: PropTypes.oneOf(['square', '16-9'])
}

Card.defaultProps = {
  title: '',
  ratio: '16-9',
  width: 350,
  height: 380,
};

export default Card;
