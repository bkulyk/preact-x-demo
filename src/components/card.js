import React from 'react';
import PropTypes from 'prop-types';

const CardMedia = ({ title, subtitle, content, media, height, width, children }) => (
  <div className="mdc-card" style={{ height, width }}>
    <div className="mdc-card__primary-action">
      {media}
    </div>
    <div className="demo-card__primary" style={{ padding: '0px 16px' }}>
      <h2 className="demo-card__title mdc-typography mdc-typography--headline6">{title}</h2>
      <h3 className="demo-card__subtitle mdc-typography mdc-typography--subtitle2">{subtitle}</h3>
    </div>
    <div className="demo-card__secondary mdc-typography mdc-typography--body2" style={{ padding: '0px 16px 8px 16px' }}>
      {content}
    </div>
    {children}
  </div>
);

CardMedia.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string,
  content: PropTypes.node,
  media: PropTypes.string,
  width: PropTypes.number,
  height: PropTypes.number,
  media: PropTypes.node,
  children: PropTypes.node,
};

CardMedia.defaultProps = {
  media: null,
  title: '',
  subtitle: '',
  content: '',
  width: 350,
  height: 'auto',
  media: '',
  children: '',
};

export default CardMedia;
