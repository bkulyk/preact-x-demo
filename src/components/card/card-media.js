import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import "@material/card/dist/mdc.card.min.css";

const Card = ({ title, url, ratio }) => (
  <div className="mdc-card__primary-action">
    <div className={`mdc-card__media mdc-card__media--${ratio}`} style={{ backgroundImage: `url(${url})` }}>
      <div className="mdc-card__media-content">{title}</div>
    </div>
  </div>
);

Card.propTypes = {
  title: PropTypes.string,
  url: PropTypes.string,
  ratio: PropTypes.oneOf(['square', '16-9'])
}

Card.defaultProps = {
  title: '',
  ratio: '16-9',
  width: 350,
  height: 380,
};

export default Card;
