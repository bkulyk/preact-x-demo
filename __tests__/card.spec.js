import React from 'react';
import { default as Card } from '@bitchin/card';
import { shallow } from './enzyme';

it('render with some defaults', () => {
  expect(shallow(<Card />)).toMatchSnapshot();
});

it('should accept a bunch of props', () => {
  expect(shallow((
    <Card
      title="Star Wars IV"
      subtitle="A New Hope"
      media="-media-placeholder-"
      content="-content-placeholder-"
      children="-children-placeholder-"
    />
  ))).toMatchSnapshot();
});