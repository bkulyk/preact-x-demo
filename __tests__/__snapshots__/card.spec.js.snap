// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with some defaults 1`] = `
<div
  className="mdc-card"
  style={
    Object {
      "height": "auto",
      "width": "350px",
    }
  }
>
  <div
    className="mdc-card__primary-action"
  />
  <div
    className="demo-card__primary"
    style={
      Object {
        "padding": "0px 16px",
      }
    }
  >
    <Typography
      className="card__title"
      component="h2"
      use="headline6"
    />
    <Typography
      className="card__subtitle"
      component="h3"
      use="subtitle2"
    />
  </div>
  <Typography
    className="demo-card__secondary"
    component="div"
    style={
      Object {
        "padding": "0px 16px 8px 16px",
      }
    }
    use="body2"
  />
</div>
`;

exports[`should accept a bunch of props 1`] = `
<div
  className="mdc-card"
  style={
    Object {
      "height": "auto",
      "width": "350px",
    }
  }
>
  <div
    className="mdc-card__primary-action"
  >
    -media-placeholder-
  </div>
  <div
    className="demo-card__primary"
    style={
      Object {
        "padding": "0px 16px",
      }
    }
  >
    <Typography
      className="card__title"
      component="h2"
      use="headline6"
    >
      Star Wars IV
    </Typography>
    <Typography
      className="card__subtitle"
      component="h3"
      use="subtitle2"
    >
      A New Hope
    </Typography>
  </div>
  <Typography
    className="demo-card__secondary"
    component="div"
    style={
      Object {
        "padding": "0px 16px 8px 16px",
      }
    }
    use="body2"
  >
    -content-placeholder-
  </Typography>
  -children-placeholder-
</div>
`;
